airflow/executors/dask_executor.py

```diff
@@ -84,15 +91,18 @@ def _process_future(self, future):
                 self.success(key)
             self.futures.pop(future)
 
-    def sync(self):
+    def sync(self) -> None:
+        if not self.futures:
+            raise AirflowException(NOT_STARTED_MESSAGE)
         # make a copy so futures can be popped during iteration
         for future in self.futures.copy():
             self._process_future(future)
 
-    def end(self):
-        for future in distributed.as_completed(self.futures.copy()):
+    def end(self) -> None:
+        if not self.client:
+            raise AirflowException(NOT_STARTED_MESSAGE)
+        if not self.futures:
+            raise AirflowException(NOT_STARTED_MESSAGE)
+        self.client.cancel(list(self.futures.keys()))
+        for future in as_completed(self.futures.copy()):
             self._process_future(future)
-
-    def terminate(self):
```

Comment:
Looks like API change: https://github.com/apache/airflow/pull/6596/files#r352302966

Reply:
Re-added it back even if it was not used in the latest push.  I think Executors were not meant to be extended (or at least with the thought that they might need to be adjusted in the future version of airflow) but I brought it back anyway.